sky/data/skystorage.py

```diff
@@ -0,0 +1,334 @@
+"""Skystorage module"""
+import os
+import pathlib
+import subprocess
+import time
+from typing import Any, List, Optional
+import urllib.parse
+
+import click
+import psutil
+
+from sky import sky_logging
+from sky.utils import common_utils
+from sky.utils import db_utils
+
+logger = sky_logging.init_logger(__name__)
+
+_CSYNC_BASE_PATH = '~/.skystorage'
+
+_DB_PATH = os.path.expanduser('~/.sky/skystorage.db')
+pathlib.Path(_DB_PATH).parents[0].mkdir(parents=True, exist_ok=True)
+
+
+def create_table(cursor, conn):
+    cursor.execute("""\
+        CREATE TABLE IF NOT EXISTS running_csync (
+        csync_pid INTEGER PRIMARY KEY,
+        sync_pid INTEGER DEFAULT 0,
```

Comment:
`-1` is more commonly used for non-exist field.

Also, we probably want to add an additional `boottime` field to note down the `psutil.boot_time()`, and whenever we check the database, we should only consider the entries with the same `boot_time`, to avoid staled entries to cause problems.

Reply:
@Michaelvll I updated in a way that `_BOOT_TIME` is first inserted when a row is created within `_add_running_csync`. Also, it now checks `_BOOT_TIME` whenever any `_get_` method is ran.

Reply:
Updated from `0` to `-1` as well